### PR TITLE
Reorganize String Algorithms

### PR DESCRIPTION
**Pull request by Kalkwst**
- [ ] I have performed a self-review of my code
- [ ] My code follows the style guidelines of this project
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Strings/PatternMatching/GeneralStringAlgorithms.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 using System;
 
-namespace Algorithms.Strings
+namespace Algorithms.Strings.PatternMatching
```
Comment (siriak) [inline Algorithms/Strings/PatternMatching/GeneralStringAlgorithms.cs:3]: This is not pattern matching. Could you just leave it in Strings? I can't come up with a better structure

**Review by siriak — CHANGES_REQUESTED** [review]
Looks good, I only have 1 comment

File: Algorithms/Strings/PatternMatching/GeneralStringAlgorithms.cs (hunk 1 above)
Comment (Kalkwst) [reply Algorithms/Strings/PatternMatching/GeneralStringAlgorithms.cs:3]: Internally it implements a LongestConsecutiveCharacter algorithm, which could be classified as Pattern Matching, but in general, yes, it is not very clear what or how this class can be organized. I moved it out of the Pattern Matching directory.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**